node/src/types/appendable_block.rs

```diff
@@ -56,26 +73,48 @@ impl AppendableBlock {
     pub(crate) fn new(transaction_config: TransactionConfig, timestamp: Timestamp) -> Self {
         AppendableBlock {
             transaction_config,
-            deploys: Vec::new(),
-            transfers: Vec::new(),
+            transactions: Vec::new(),
             timestamp,
-            deploy_and_transfer_set: HashSet::new(),
+            transaction_and_transfer_set: HashSet::new(),
             total_gas: Gas::zero(),
             total_size: 0,
             total_approvals: 0,
         }
     }
 
-    /// Attempts to add any kind of deploy (transfer or other kind).
+    /// Attempts to add any kind of transaction (transfer or other kind).
     pub(crate) fn add(
         &mut self,
-        deploy_hash_with_approvals: DeployHashWithApprovals,
-        footprint: &DeployFootprint,
+        transaction_hash_with_approvals: TransactionHashWithApprovals,
+        footprint: &TransactionFootprint,
     ) -> Result<(), AddError> {
-        if footprint.is_transfer {
-            self.add_transfer(deploy_hash_with_approvals, footprint)
-        } else {
-            self.add_deploy(deploy_hash_with_approvals, footprint)
+        match (transaction_hash_with_approvals, footprint) {
```

Comment:
Most of the implementations of `add_transfer`, `add_deploy` and `add_transaction_v1` are the same. Maybe we can group the common parts into a common function such that we don't need to deconstruct and re-construct the transaction hash every time.